sycl/include/CL/sycl/aliases.hpp

```diff
@@ -0,0 +1,109 @@
+//==----------- aliases.hpp --- SYCL type aliases --------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/common.hpp>
+
+#include <cstddef>
+#include <cstdint>
+
+namespace cl {
+namespace sycl {
+template <typename T, int N> class vec;
+namespace detail {
+namespace half_impl {
+class half;
+} // namespace half_impl
+} // namespace detail
+} // namespace sycl
```

Comment:
I think you can simplify the code a little bit by not closing namespaces `sycl` and `cl` here - anyway you are declaring them again later in this file

Reply:
These namespaces are closed to declare half in the global namespace.
I can certainly move the half declaration in global namespace to the eof, but I think that in at the beginning of the file more amply.